front/lib/resources/storage/models/group_memberships.ts

```diff
@@ -0,0 +1,112 @@
+import type {
+  CreationOptional,
+  ForeignKey,
+  InferAttributes,
+  InferCreationAttributes,
+} from "sequelize";
+import { DataTypes, Model, Op } from "sequelize";
+
+import { User } from "@app/lib/models/user";
+import { Workspace } from "@app/lib/models/workspace";
+import { frontSequelize } from "@app/lib/resources/storage";
+import { GroupModel } from "@app/lib/resources/storage/models/groups";
+
+export class GroupMembershipModel extends Model<
+  InferAttributes<GroupMembershipModel>,
+  InferCreationAttributes<GroupMembershipModel>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+
+  declare startAt: Date;
+  declare endAt: Date | null;
+
+  declare groupId: ForeignKey<GroupModel["id"]>;
+  declare userId: ForeignKey<User["id"]>;
+  declare workspaceId: ForeignKey<Workspace["id"]>;
+
+  static async checkOverlap(
```

Comment:
I'm not a big fan of adding logic in Sequelize it will make the transition even harder. Can we use a unique index 
 on workspaceId, userId, groupId, here?

Reply:
>  Can we use a unique index on workspaceId, userId, groupId, here?

We want to be able to belong to a group, then not, then again though

Reply:
Yeah but here group is required, right?

Reply:
Agreed I wold rather have this in the resource vs validate in sequelize :+1:

Reply:
The argument to not have it in sequelize is that it reduce our dependency on it simplifying our life if/when we migrate off of it?

Reply:
I would not say this was making our life complex but I added a commit to remove it 😄